### PR TITLE
Document the release process and update the "release-it" options

### PR DESCRIPTION
**Pull request by ttmc**
This is to help with issue #292 

- Added a `RELEASE_PROCESS.md` file based on:
  - the [BigchainDB Server release process](https://github.com/bigchaindb/bigchaindb/blob/master/RELEASE_PROCESS.md)
  - the [oceanprotocol/commons release process](https://github.com/oceanprotocol/commons#%EF%B8%8F-releases)
- Updated the command-line options passed to `release-it` based on the same options in oceanprotol/commons, i.e. in `package.json`

I hope the process is mostly correct. I wasn't able to test it. Please take a look to make sure it makes sense. Once someone uses it, they might find something that should be changed or clarified.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,9 +23,9 @@
     "clean": "rimraf dist/bundle dist/node",
     "test": "npm run lint && nyc ava test/ && npm run thanks && npm run report-coverage",
     "thanks": "cowsay Hi, thanks for your interest in BigchainDB. We appreciate your contribution!",
-    "release": "./node_modules/release-it/bin/release-it.js --src.tagName='v%s' --github.release --npm.publish --non-interactive",
-    "release-minor": "./node_modules/release-it/bin/release-it.js minor --src.tagName='v%s' --github.release --npm.publish --non-interactive",
-    "release-major": "./node_modules/release-it/bin/release-it.js major --src.tagName='v%s' --github.release --npm.publish --non-interactive",
+    "release": "./node_modules/release-it/bin/release-it.js --git.tagName='v${version}' --github.release --npm.publish --non-interactive",
```
Comment (kremalicious) [inline package.json]: I would assume the `release-it` package also needs to be updated so those new commands work. Current version is `12.2.0` https://www.npmjs.com/package/release-it

Comment (ttmc) [reply package.json]: Yes, one of the steps in `RELEASE_PROCESS.md` is:

> We use the [release-it](https://www.npmjs.com/package/release-it) package (from npm) to automate most of the release. Make sure you have a recent version.

Comment (kremalicious) [reply package.json]: recent version needs to be defined in the package.json. That's where `release-it` is coming from after doing `npm install`

Comment (ttmc) [reply package.json]: Okay I updated `release-it` to 12.2.0 in `package.json`.

Comment (kremalicious) [reply package.json]: 👍

**Comment by ttmc** [comment]
I realized I forgot to include the signed-off-by line in the Git commit messages, so I will see if I can fix those.

**Comment by ttmc** [comment]
I had to do a force-push of the revised commits, but it looks like it worked. Disaster averted!

**Comment by eckelj** [comment]
Perfect. Will review asap I do have a minute

**Comment by manolodewiner** [comment]
`release-it` latest version is now 12.4.1 but everything should work as well with 12.2.0.
Since a lot of packages needs to be updated this can be done in another PR, therefore I will approve it.

**Pull request merged**